### PR TITLE
[Left TODO] Support PaddleSeg deployment

### PR DESCRIPTION
**Pull request by felixhjh**
1. 支持 PaddleSeg 六个模型的部署（Deeplabv3-ResNet50, hrnet-w18, pp-humanseg, pp-humanseg-server, PP-LiteSeg, unet）
2. 支持三种模型导出形式（logits matrix, logits matrix with argmax, logits matrix with argmax and with softmax）
3. 支持visualization

### CODE REVIEW

File: csrcs/fastdeploy/vision/ppseg/model.cc [hunk 1]
```diff
@@ -74,49 +75,185 @@ bool Model::BuildPreprocessPipelineFromConfig() {
...
-bool Model::Postprocess(const FDTensor& infer_result,
-                        SegmentationResult* result) {
-  FDASSERT(infer_result.dtype == FDDataType::INT64,
-           "Require the data type of output is int64, but now it's " +
+bool Model::Postprocess(FDTensor& infer_result, SegmentationResult* result,
+                        std::map<std::string, std::array<int, 2>>* im_info) {
+  FDASSERT(infer_result.dtype == FDDataType::INT64 ||
+               infer_result.dtype == FDDataType::FP32,
+           "Require the data type of output is int64 or fp32, but now it's " +
                Str(const_cast<fastdeploy::FDDataType&>(infer_result.dtype)) +
```
Comment (jiangjiajun) [inline csrcs/fastdeploy/vision/ppseg/model.cc]: 这里是不是可以直接写成`Str(infer_result.dtype)`

File: csrcs/fastdeploy/vision/ppseg/model.cc [hunk 2]
```diff
@@ -74,49 +75,185 @@ bool Model::BuildPreprocessPipelineFromConfig() {
...
+      for (int c = 0; c < channel; c++) {
+        for (int h = 0; h < height; h++) {
+          for (int w = 0; w < width; w++) {
+            hwc_data[n * chw + h * wc + w * channel + c] =
+                *(infer_result_buffer + index);
+            index++;
+          }
+        }
+      }
+    }
```
Comment (jiangjiajun) [inline csrcs/fastdeploy/vision/ppseg/model.cc]: 这个hwc2chw是不是可以直接改成opencv的操作

File: csrcs/fastdeploy/vision/ppseg/model.cc [hunk 3]
```diff
@@ -74,49 +75,185 @@ bool Model::BuildPreprocessPipelineFromConfig() {
...
+    if (infer_result.dtype == FDDataType::INT64) {
+      int64_t chw = channel * height * width;
+      int64_t* infer_result_buffer = static_cast<int64_t*>(infer_result.Data());
+      std::vector<float_t> float_result_buffer(chw);
+      temp_mat = cv::Mat(height, width, CV_32FC(channel));
+      int index = 0;
+      for (int i = 0; i < height; i++) {
+        for (int j = 0; j < width; j++) {
+          temp_mat.at<float_t>(i, j) =
+              static_cast<float_t>(infer_result_buffer[index++]);
```
Comment (jiangjiajun) [inline csrcs/fastdeploy/vision/ppseg/model.cc]: processor的这个函数建议按照功能模块拆一下，提升可读性

**Review by jiangjiajun — CHANGES_REQUESTED** [review]

File: csrcs/fastdeploy/vision/ppseg/model.cc [hunk 4]
```diff
@@ -74,49 +75,185 @@ bool Model::BuildPreprocessPipelineFromConfig() {
...
+
+    if (result->contain_score_map) {
+      channel = infer_result.shape[3];
+    }
+    cv::Mat temp_mat;
+    if (infer_result.dtype == FDDataType::INT64) {
+      int64_t chw = channel * height * width;
+      int64_t* infer_result_buffer = static_cast<int64_t*>(infer_result.Data());
+      std::vector<float_t> float_result_buffer(chw);
+      temp_mat = cv::Mat(height, width, CV_32FC(channel));
```
Comment (jiangjiajun) [inline csrcs/fastdeploy/vision/ppseg/model.cc]: int64出来的结果，转为了float的来进行后面的resize处理，这个部署的结果是否跟PaddleSeg对齐了？

**Review by jiangjiajun — CHANGES_REQUESTED** [review]

File: csrcs/fastdeploy/vision/ppseg/model.cc (hunk 4 above)
Comment (felixhjh) [reply csrcs/fastdeploy/vision/ppseg/model.cc]: 这么做的原因有二：
1. int64的结果出来的是类别的mask矩阵，对于pp-human系列也就是0,1矩阵（其他也就是0-num_classes的矩阵）。opencv没有64位的矩阵变量，因此必须要转。
2. 转为float是因为，resize的插值方式双线性策略（inter=1，也是前处理所使用的策略）只支持float类型opencv mat（不支持int、uint8精度缺失）。为保持前后处理一致所以统一转为float。
3. PaddleSeg给到的三张示例图片，正常人像、竖屏人像、横屏人像fastdeploy结果与PaddleSeg是一致的，后续evaluation模块开发完成（正在去除Paddle依赖）会跑全量

Comment (jiangjiajun) [reply csrcs/fastdeploy/vision/ppseg/model.cc]: 第3点， 先确认输出的值(numpy) 与 PaddleSeg是一致的，并做到与Seg结果一致

**Review by jiangjiajun — APPROVED** [review]

**Comment by jiangjiajun** [comment]
- WARNING日志的关闭，当前在infer里面每次都会输出提示，需让用户能够关闭
- NHWC2CHW的实现应该为FDTensor的基础操作，需集成Eigen库来解决，后续需替换
